python/lsst/pipe/supertask/config.py

```diff
@@ -47,6 +47,50 @@
 # ------------------------
 
 
+def InputDatasetField(doc, name, units, storageClass, check=None):
+    """Factory function to create `~lsst.pex.config.Config` class instances
+    of `InputDataSetConfig`
+
+    This function servers as syntactic sugar for creating Configurable fields
+    which are `InputDatasetConfigs`. The naming of this function violates the
+    normal convention of a lowercase first letter in the function name, as
+    this function is intended to sit in the same place as
+    `~lsst.pex.config.ConfigField` classes, and consistency in declaration
+    syntax is important.
+
+    The input arguments for this class are a combination of the arguments for
+    `~lsst.pex.config.ConfigField` and `InputDatasetConfig`. The arguments
+    doc and check come from `~lsst.pex.config.ConfigField`, while name,
+    units, and storageClass come from `InputDatasetConfig`.
+
+    Parameters
+    ----------
+    doc : `str`
+        Documentation string for the `InputDatasetConfig`
+    name : `str`
+        Name of the `~lsst.daf.butler.DatasetType` in the returned
+        `InputDatasetConfig`
+    units : iterable
```

Comment:
either say:
```
units : `list` of `~lsst.daf.butler.DataUnit`
```
or say
```
units : sequence of `~lsst.daf.butler.DataUnit`
```
since we are declaring the contents of the list.

Reply:
must it be a list though? A pex config ListField will happily take any iterable, as long as the data type inside is the field's declared data type. So the second is more preferable than the first, but I am not sure why iterable of `str` is not. This may be my misunderstanding, but I thought the string after the colon was supposed to be a 'type' and any additional information about the type would go in the documentation line, hence why I just had it listed as iterable.

Reply:
numpydoc allows you to describe the container and the thing it contains. Hence 
```
unit : iterable of `str`
   Iterable of DataUnit strings...
```
seems reasonable. My point was that you said iterable but not what was in it, and then in the description you said a list. Is an iterable okay or do we mean sequence? ie is the API guaranteeing it only reads through the iterable once?